### PR TITLE
INGK-1012 Drive desconnection not detected when PDO poller is active

### PR DESCRIPTION
**Pull request by MartinALL91**
Fixes # INGK-1012

### Type of change

- Upon stopping the PDOs, only reset the slave states to PreOp if the slaves are connected and in either the Op, SafeOp, or SafeOp + Error states.

### Tests
Test 1:
- Connect to ML3.
- Run the PDO poller.
- Turn off the power supply.
- Check that the drive disconnection is detected.
- Turn on the power supply.
- Check that the drive re-connects.
- Run the PDO poller and check that it works.

Test 2:
- Connect to ML3
- Run the PDO poller.
- Disconnect the ethernet cable.
- Check that the drive disconnection is detected.
- Connect the ethernet cable.
- Check that the drive re-connects.
- Run the PDO poller and check that it works.

### Documentation

- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting and linting

- [x] Use the ruff package to format the code: `ruff format ingenialink tests virtual_drive`.
- [x] Use the ruff package to lint the code: `ruff check ingenialink tests virtual_drive`.

### Others

- [x] Set fix version field in the Jira issue.

### CODE REVIEW

File: ingenialink/ethercat/network.py [hunk 1]
```diff
@@ -295,6 +295,8 @@ def stop_pdos(self) -> None:
             for servo in self.servos
             if servo.slave.state in [pysoem.OP_STATE, pysoem.SAFEOP_STATE]
         ]
+        if len(op_servo_list) == 0:
```
Comment (polfeliu) [inline ingenialink/ethercat/network.py:298]: Is this change related?

Comment (MartinALL91) [reply ingenialink/ethercat/network.py:298]: Yes, if no drives are in the OP or SafeOp state (they're in PreOp or disconnected), there is no need to reset them (transition back to PreOp).

**Review by GullinBustin — APPROVED** [review]

**Review by polfeliu — APPROVED** [review]
